taxcalc/tests/test_calculate.py

```diff
@@ -285,12 +286,47 @@ def test_diagnostic_table():
     tax_dta.flpdyr += 17
     puf = Records(tax_dta, weights = weights)
     # Create a Calculator
-    
+
     calc = Calculator(params=params, records=puf, sync_years=False)
 
     calc.diagnostic_table()
 
 
+def test_aggregate_corporate_income_tax_vals():
+    # we need the records' year at 2008 for blow up step. So param's year needs to be 2008 to past the test
+    irates = {2008:0.015, 2009:0.020, 2010:0.022, 2011:0.020, 2012:0.021,
+          2013:0.022, 2014:0.023, 2015:0.024, 2016:0.024, 2017:0.024,
+          2018:0.024, 2019:0.024}
+    params = Parameters(start_year=2008, inflation_rates=irates)
+    # Create a Public Use File object
+    puf = Records(tax_dta)
+    # Create a Calculator
+    calc = Calculator(params=params, records=puf)
+
+    agg_e_and_p = ((calc.records.e02000 * calc.records.s006).sum()
+                   + (calc.records.e09400 * calc.records.s006).sum())
+
+    agg_dividends = (calc.records.e00600 * calc.records.s006).sum()
+
+    agg_capgains = ((calc.records.e23250 * calc.records.s006).sum() +
+                    (calc.records.e22250 * calc.records.s006).sum() +
+                    (calc.records.e23660 * calc.records.s006).sum())
+
+    agg_bonds = ((calc.records.e00400 * calc.records.s006).sum()
+                 + (calc.records.e00300 * calc.records.s006).sum())
+
+    agg_compensation = ((calc.records.e00200 * calc.records.s006).sum() +
+                        (calc.records.e00250 * calc.records.s006).sum() +
+                        (calc.records.e33420 * calc.records.s006).sum() +
+                        (calc.records.e03600 * calc.records.s006).sum() +
+                        (calc.records.e07240 * calc.records.s006).sum())
+
+    assert abs(calc.records.agg_comp - agg_compensation) < .001
```

Comment:
On Thu, 2 Jul 2015, T.J. Alumbaugh wrote:

> 
> In taxcalc/tests/test_calculate.py:
> 
> > +    agg_dividends = (calc.records.e00600 * calc.records.s006).sum()
> > +
> > +    agg_capgains = ((calc.records.e23250 * calc.records.s006).sum() +
> > +                    (calc.records.e22250 * calc.records.s006).sum() +
> > +                    (calc.records.e23660 * calc.records.s006).sum())
> > +
> > +    agg_bonds = ((calc.records.e00400 * calc.records.s006).sum()
> > +                 + (calc.records.e00300 * calc.records.s006).sum())
> > +
> > +    agg_compensation = ((calc.records.e00200 * calc.records.s006).sum() +
> > +                        (calc.records.e00250 * calc.records.s006).sum() +
> > +                        (calc.records.e33420 * calc.records.s006).sum() +
> > +                        (calc.records.e03600 * calc.records.s006).sum() +
> > +                        (calc.records.e07240 * calc.records.s006).sum())
> > +
> > +    assert abs(calc.records.agg_comp - agg_compensation) < .001
> 
> As far as I know, none of these aggregates are in the code at all right now. It looks like this
> is a proposed addition. I was going to ask @MattHJensen if this was a feature for the web app.
> If we accept these changes, the standard behavior would be to calculate these values for each
> year. We could set some kind of flag to indicate whether or not to do these calculations, but if
> you start down that path, it can get messy pretty quickly.
> 
> —
> Reply to this email directly or view it on
> GitHub.[AHvQVVW6uim2zfFzMWR5c39R1mzGXvTIks5oZThtgaJpZM4FES6E.gif]
>

These don't change with changes to planx or plany, do they? We could run 
it once and post the aggregates on a static page, if we feel it useful
to report these numbers.

dan

> 
>